client/my-sites/plans/main.jsx

```diff
@@ -225,6 +226,11 @@ class Plans extends Component {
 		);
 	};
 
+	handleSpecialOfferClick = async ( cartItems ) => {
```

Comment:
Yeah I would really like this but the `<Banner>` component doesn't currently support a `busy` or `isLoading` prop.

I'll add it. I think it's worth it since it takes a little time to add the offer to the cart before navigating.

Reply:
We can use `replaceProductsInCart` which will clear the existing cart. [Docs](https://github.com/Automattic/wp-calypso/tree/trunk/packages/shopping-cart#:~:text=replaceproductsincart).

Reply:
I just changed this as I think it's always what we want when you click the banner.

Reply:
I thought it might be a bit rude to clear their cart but you could be right. If it's just replacing a plan bundle then that seems fine. But cases I'm worried about would be if they user had found and added a domain/premium theme they really linked to the cart. They'd have to go and find them again and that seems annoying.

Reply:
I feel like the pros outweigh the cons here. We avoid issues with double-clicks and existing plans in the cart and also keep the cart items all having a single currency when the user's normal currency is not USD. I created an [instruction doc](https://docs.google.com/document/d/1pUsGKjQaB4fI_6t-CgMDNUzEi5QM4hU0vEedr0t9fGs/edit?usp=sharing) for these modified flows and have added a note about clearing the cart there so it won't be as rude.

Reply:
Ok you've convinced me. The single currency issues can be annoying. If for some reason lots of users are adding items like domains to the cart first then we can change it back later.